### ISSUE TITLE
Multiple Animated GIFs plays slow

### ISSUE THREAD

**Issue by WopKatan** [facebook/fresco#398]
Using multiple SimpleDraweeView in a gridView, displaying GIF from res, slows animation down.
it feels like there's a single thread managing all the frame counts, instead of having each view manage its own count.

Note: when displaying about 4 GIF's simultaneous - everything seems to work great. 
At about 8 SimpleDraweeView in same screen - all the GIFs plays with lags.

how can I bypass that?

some code:

```
    DraweeController controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.a))
            .setAutoPlayAnimations(true)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view1);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.b))
            .setControllerListener(controllerListener)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view2);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.c))
            .setControllerListener(controllerListener)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view3);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.d))
            .setControllerListener(controllerListener)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view4);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.e))
            .setControllerListener(controllerListener)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view5);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.f))
            .setAutoPlayAnimations(true)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view6);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.g))
            .setAutoPlayAnimations(true)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view7);
    draweeView.setController(controller);

    controller = Fresco.newDraweeControllerBuilder()
            .setUri(Uri.parse("res:///" + R.drawable.h))
            .setAutoPlayAnimations(true)
            .build();
    draweeView = (SimpleDraweeView) findViewById(R.id.my_image_view8);
    draweeView.setController(controller);
```

**Comment by storix** [comment]
I have a similar issue with Fresco v1.2.0. Multiple GIFs in `RecyclerView` are being decoded very slowly (it feels like 1 frame per sec).

I tried to supply my own `ExecutorSupplier` with decode thread count set to 10:

```
public class AnimationExecutorSupplier implements ExecutorSupplier {
    private static final int NUM_IO_BOUND_THREADS = 2;
    private static final int NUM_LIGHTWEIGHT_BACKGROUND_THREADS = 1;
    private final Executor mIoBoundExecutor;
    private final Executor mDecodeExecutor;
    private final Executor mBackgroundExecutor;
    private final Executor mLightWeightBackgroundExecutor;
    public AnimationExecutorSupplier(int cpuBoundThreadNumber, int decodingThreadNumber) {
        ThreadFactory backgroundPriorityThreadFactory =
                new PriorityThreadFactory(Process.THREAD_PRIORITY_BACKGROUND);

        mIoBoundExecutor = Executors.newFixedThreadPool(NUM_IO_BOUND_THREADS);
        mDecodeExecutor = Executors.newFixedThreadPool(
                decodingThreadNumber,
                backgroundPriorityThreadFactory);
        mBackgroundExecutor = Executors.newFixedThreadPool(
                cpuBoundThreadNumber,
                backgroundPriorityThreadFactory);
        mLightWeightBackgroundExecutor = Executors.newFixedThreadPool(
                NUM_LIGHTWEIGHT_BACKGROUND_THREADS,
                backgroundPriorityThreadFactory);
    }

    @Override
    public Executor forLocalStorageRead() {
        return mIoBoundExecutor;
    }

    @Override
    public Executor forLocalStorageWrite() {
        return mIoBoundExecutor;
    }

    @Override
    public Executor forDecode() {
        return mDecodeExecutor;
    }

    @Override
    public Executor forBackgroundTasks() {
        return mBackgroundExecutor;
    }

    @Override
    public Executor forLightweightBackgroundTasks() {
        return mLightWeightBackgroundExecutor;
    }
}
```

However, this doesn't change anything.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs that is way more performant, especially in the case where you display multiple GIFs. Let us know if you're still experiencing problems with 1.4.0.
